### PR TITLE
Add TypeScript types to some event handlers

### PR DESCRIPTION
**Pull request by bengry**
#### What does this PR do?
Adds typings for a few components' event handlers.

#### Where should the reviewer start?
Whichever, the commits are not based off of one another, and each should stand on its own.
I do suggest reviewing them one-by-one, since there's no technical relation between them.

#### What testing has been done on this PR?
A local TypeScript project, and testing the the expected types appear in a local IDE in the tests file for each component.

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/3165

Please update https://github.com/grommet/grommet/issues/3165#issuecomment-699111817 once (and if) this is merged.

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Its backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
@jasnross can you help to review this PR?

File: src/js/components/Drop/index.d.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export interface DropProps {
     left?: 'left' | 'right';
   };
   elevation?: ElevationType;
-  onClickOutside?: (...args: any[]) => any;
+  onClickOutside?: (this: Document, ev: MouseEvent) => any;
```
Comment (ShimiSun) [inline src/js/components/Drop/index.d.ts:12]: Verified on the Drop 'Multiple' story, and confirm it works. 
Should the `any` be changed to `void`?

File: src/js/components/Layer/index.d.ts [hunk 2]
```diff
@@ -18,7 +18,9 @@ export interface LayerProps {
   full?: boolean | 'vertical' | 'horizontal';
   margin?: MarginType;
   modal?: boolean;
-  onClickOutside?: (...args: any[]) => any;
+  onClickOutside?: (
```
Comment (ShimiSun) [inline src/js/components/Layer/index.d.ts:21]: Verified the behavior remains intact for the Layer 'Center' story.

File: src/js/components/Drop/index.d.ts (hunk 1 above)
Comment (bengry) [reply src/js/components/Drop/index.d.ts]: Fixed in latest revision.

**Comment by jasnross** [comment]
> @jasnross can you help to review this PR?

Sure, I'll take a look a bit later today!

**Comment by bengry** [comment]
> > @jasnross can you help to review this PR?
> 
> Sure, I'll take a look a bit later today!

Any news on this? I just rebased my branch on top of the latest `master`.

**Review by jcfilben — COMMENTED** [review]
Looks good, I tested DropButton onClose and onOpen with Simple.tsx story. Tested Drop onClickOutside with Multiple.tsx and Progressive.tsx. Tested Layer onClickOutside with Target.tsx, Center.tsx, Corner.tsx, and Form.tsx.

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Toda @bengry, and thank you for your contributions! 💟

**Pull request merged**